### ISSUE TITLE
BuildAnalyzer should analyze files as a single batch to handle external JS properly

### ISSUE THREAD

**Issue by rictic** [Polymer/polymer-build#179]
Simple repro:

index.html
```html
<script>
  /** @polymerBehavior */
  var Behave = {};
</script>
<script src="./element.js"></script>
```

-- element.js
```js
Polymer({is: 'my-element', behaviors: [Behave]});
```

This should not result in any warnings, but currently it does. The reason is that if you analyze files individually, `element.js` has no way to know that it is included in `index.html` and can use `index.html` to help resolve its behaviors. In a future version of the analyzer we'll be able to solve this so long as the analyzer sees `index.html` first, but for now the only solution is to analyze all the files that you're going to analyze all at once.

So when we update to analyzer alpha.37 we should update BuildAnalyzer to buffer up files and perform one analyze call with the array of sources.

**Comment by FredKSchott** [comment]
That's interesting, I don't know how I feel about keeping the shared `window` state across analyzed documents, and having the order of analysis affecting the results of analysis. But without an actual module loading system I understand that our hands are tied. Still, I'd much prefer explicit dependency imports, maybe something similar to TS's `/// <reference path="... />` annotations?

In regards to build, i don't know if this is necessary either way. We only analyze entrypoints directly, and use the dependencies that the analyzer loads during it's analysis to populate the `dependencies()` stream. We never call `analyze()` on non-entrypoints as far as I know.

**Comment by rictic** [comment]
Yeah, we'd all much prefer explicit dependencies, but the user experience of needing to add the back-link is fairly miserable. _Maybe_ it would be ok to require it if we had an automatic `polymer lint --fix` step that added the backrefs.

If we only call `analyze` on html files here then we should be fine. I think we should add a regression test to be sure though.
